### PR TITLE
[mono][wasm] MacOSX Catalina Beta wasm workaround for booting mono wasm from Blazor.

### PR DESCRIPTION
**Pull request by kjpou1**
Summary of the changes (Less than 80 chars)
 - MacOSX Catalina initializes WebAssembly `Module` differently and needs to be initialized before loading `mono.js`.

Modifications:

 - Add new method to MonoPlatform.ts `addGlobalModuleScriptTagsToDocument` 
   - Make sure we have the Module defined from the page as a `<script></script>` tag
   - When the `Module` is defined the regular Blazor mono webassembly setup is followed.
  - Boot start modified to call the new method `addGlobalModuleScriptTagsToDocument`.
 
After modification:

- New `<script></script>` is added to the page that defines the WebAssembly `Module` before the addition of the `<script></script>` tag to load `mono.js`.

### CODE REVIEW

File: src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts [hunk 1]
```diff
@@ -26,9 +26,11 @@ export const monoPlatform: Platform = {
         init: () => { },
       };
       // Emscripten works by expecting the module config to be a global
-      window['Module'] = createEmscriptenModuleInstance(loadAssemblyUrls, resolve, reject);
-
-      addScriptTagsToDocument();
+      // MacOSX Catalina initializes Module differently so needa workaround for now.
```
Comment (jaykrell) [inline src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts]: "needa" is not a word.

File: src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts [hunk 2]
```diff
@@ -205,6 +207,24 @@ function addScriptTagsToDocument() {
...
+// This may not be necassary in the Catalina release.
+function addGlobalModuleScriptTagsToDocument(callback: () => void) {
+  const browserSupportsNativeWebAssembly = typeof WebAssembly !== 'undefined' && WebAssembly.validate;
+  if (!browserSupportsNativeWebAssembly) {
+    throw new Error('This browser does not support WebAssembly.');
+  }
+
+  const scriptElem = document.createElement('script');
+
+  // This polutes global but is needed so it can be called from the script.
```
Comment (jaykrell) [inline src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts]: "pollutes"

**Comment by SteveSandersonMS** [comment]
Thanks very much for this, @kjpou1!

Before I proceed with merging this, could you please comment about this comment:

```
// Make sure we have the Module setup from the page due to a difference in MacOSX Catalina Beta.
// This may not be necassary in the Catalina release.
```

Now that iOS 13 (and even 13.1) have shipped, could you confirm whether or not this workaround is still needed? If it is then presumably we can update this comment not to say "may not be necessary".

**Comment by SteveSandersonMS** [comment]
Thanks again for this. We're about to merge it, so I needed to make a couple of further tweaks. So, closing this in favour of https://github.com/aspnet/AspNetCore/pull/14576 which adds those tweaks.

**Pull request closed without merging**